library/core/src/slice/iter/macros.rs

```diff
@@ -64,7 +64,7 @@ macro_rules! iterator {
         // backwards by `n`. `n` must not exceed `self.len()`.
         macro_rules! zst_shrink {
             ($self: ident, $n: ident) => {
-                $self.end = ($self.end as * $raw_mut u8).wrapping_offset(-$n) as * $raw_mut T;
+                $self.end = $self.end.wrapping_byte_offset(-$n);
```

Comment:
This change is fine, but I definitely look at it and wonder if it can be
```suggestion
                $self.end = $self.end.wrapping_byte_sub($n);
```
since it's explicitly `shrink`.

Reply:
I haven't done this originally because `($n): isize`, but I've just checked and it seems that this is only called with literals and `smt as isize` so this change makes sense.